atoMEC/xc.py

```diff
@@ -302,5 +301,87 @@ def calc_xc(density, xgrid, xcfunc, xctype):
                 xc_arr = out["zk"].transpose()[0]
             elif xctype == "v_xc":
                 xc_arr = out["vrho"].transpose()
+        # gga
+        if xcfunc._family == 2:
+            rho_libxc = np.zeros((config.grid_params["ngrid"], config.spindims))
+
+            for i in range(config.spindims):
+                rho_libxc[:, i] = density[i, :]
+            # preparing the sigma array needed for libxc gga calculation
+            if config.spindims == 2:
+                sigma_libxc = np.zeros((config.grid_params["ngrid"], 3))
+                grad_0 = mathtools.grad_den(density[0, :], np.exp(xgrid), xgrid)
+                grad_1 = mathtools.grad_den(density[1, :], np.exp(xgrid), xgrid)
+                sigma_libxc[:, 0] = grad_0 ** 2
+                sigma_libxc[:, 1] = grad_0 * grad_1
+                sigma_libxc[:, 2] = grad_1 ** 2
+            else:
+                sigma_libxc = np.zeros((config.grid_params["ngrid"], 1))
+                grad = mathtools.grad_den(density[0, :], np.exp(xgrid), xgrid)
+                sigma_libxc[:, 0] = grad ** 2
+
+            inp = {"rho": rho_libxc, "sigma": sigma_libxc}
 
+            # compute the xc potential and energy density
+            out = xcfunc.compute(inp)
+            # extract the energy density
+            if xctype == "e_xc":
+                xc_arr = out["zk"].transpose()[0]
+            # extract xc potential
+            elif xctype == "v_xc":
+                if config.spindims == 2:
+                    xc_arr = out["vrho"].transpose() - eta(out, grad_0, grad_1, xgrid)
+                else:
+                    xc_arr = np.zeros((config.grid_params["ngrid"], config.spindims))
+                    for i in range(config.spindims):
```

Comment:
Add comment to explain what's happening (something like 'use chain rule to obtain potential from libxc outputs')